lib/OpenQA/WebAPI/Controller/Main.pm

```diff
@@ -24,7 +24,9 @@ use OpenQA::Schema::Result::Jobs;
 sub _group_result {
     my ($self, $group, $limit) = @_;
 
-    my $timecond = {">" => time2str('%Y-%m-%d %H:%M:%S', time - 24 * 3600 * 14, 'UTC')};
+    my $time_limit_days = $self->param('time_limit_days') // 14;
```

Comment:
Sometimes you can really think around corners and ask about weird unusual cases. I like that :-)

Calling `http://localhost:9526/group_overview/1?time_limit_days=foo` renders an empty page and the log output:
```
[00][Thu Feb 25 11:44:16 2016] [debug] GET "/group_overview/1"
[00][Thu Feb 25 11:44:16 2016] [debug] Routing to controller "OpenQA::WebAPI::Controller::Main" and action "group_overview"
[00][Thu Feb 25 11:44:16 2016] [debug] Retrieving results for up to 10 builds up to foo days old
[00]Argument "foo" isn't numeric in multiplication (*) at
[00]    /home/okurz/local/os-autoinst/openQA/script/../lib/OpenQA/WebAPI/Controller/Main.pm line 29 (#1)
[00]    (W numeric) The indicated string was fed as an argument to an operator
[00]    that expected a numeric value instead.  If you're fortunate the message
[00]    will identify which operator was so unfortunate.
[00]    
[00][Thu Feb 25 11:44:16 2016] [debug] Rendering cached template "main/group_overview.html.ep"
[00][Thu Feb 25 11:44:16 2016] [debug] Rendering cached template "layouts/info.html.ep"
[00][Thu Feb 25 11:44:16 2016] [debug] Rendering cached template "main/group_builds.html.ep"
[00][Thu Feb 25 11:44:16 2016] [debug] Rendering cached template "layouts/bootstrap.html.ep"
[00][Thu Feb 25 11:44:16 2016] [debug] Rendering cached template "branding/openSUSE/navbar.html.ep"
[00][Thu Feb 25 11:44:16 2016] [debug] 200 OK (0.014780s, 67.659/s)
```
So I would not over-engineer something here. As long as it is only used as a URL parameter I would not really invest more time in checking this more, an empty page should be good enough. If we would provide something like a search field this would invite also "unexperienced" users to type something and we should catch bad user input. agreed?